src/types.py

```diff
@@ -0,0 +1 @@
+ReplicaInfos = list[dict[str, object]]
```

Comment:
I think it would be better to introduce a singular type `ReplicaInfo`, and use `List[ReplicaInfo]` in annotations. Or just introduce 2 types.